### PR TITLE
Implement nsxt_edge_transport_node data source

### PR DESCRIPTION
**Pull request by ksamoray**

### CODE REVIEW

File: nsxt/data_source_nsxt_transport_node.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+func dataSourceNsxtTransportNode() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNsxtTransportNodeRead,
+
+		Schema: map[string]*schema.Schema{
+			"id":           getDataSourceIDSchema(),
+			"display_name": getDataSourceExtendedDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"node_type": {
```
Comment (annakhm) [inline nsxt/data_source_nsxt_transport_node.go:24]: Would it make sense to filter by node type as well?

Comment (ksamoray) [reply nsxt/data_source_nsxt_transport_node.go:24]: Would that produce a unique result, other than in testing environments?

Comment (annakhm) [reply nsxt/data_source_nsxt_transport_node.go:24]: Seems to me like such filtering could be helpful, but we can wait for the ask

**Review by annakhm — APPROVED** [review]

File: nsxt/data_source_nsxt_transport_node.go (hunk 1 above)
Comment (ksamoray) [reply nsxt/data_source_nsxt_transport_node.go:24]: Maybe it's better to wait and see if we want to convert this into nsxt_edge_transport_node as discussed? Or maybe we change only the resource?

Comment (annakhm) [reply nsxt/data_source_nsxt_transport_node.go:24]: I don't really have a preference, either is fine I think

**Pull request merged**